cmd/cone/get_drop_task.go

```diff
@@ -49,17 +49,142 @@ func taskCmd(cmd *cobra.Command) *cobra.Command {
+	if maxProvisionTime == nil && duration == "" {
+		return nil, nil
+	}
+
+	if v.GetBool(nonInteractiveFlag) {
+		return handleDurationNonInteractive(maxProvisionTime, duration)
+	}
+	errorPrinter := pterm.PrefixPrinter{
+		MessageStyle: &pterm.ThemeDefault.ErrorMessageStyle,
+		Prefix: pterm.Prefix{
+			Style: &pterm.ThemeDefault.ErrorPrefixStyle,
+			Text:  " ERROR ",
+		},
+	}
+	input := pterm.DefaultInteractiveTextInput.WithMultiLine(false)
+	firstRun := true
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+			if !firstRun {
+				var err error
+				duration, err = input.Show()
+				if err != nil {
+					return nil, err
+				}
+			}
+			firstRun = false
+
+			formattedDuration, err := strToDur(duration)
+			if err != nil {
+				errorPrinter.Println(err.Error())
+				continue
+			}
+
+			err = validateGrantTaskArguments(maxProvisionTime, formattedDuration)
+			if err == nil {
+				return formattedDuration, nil
+			}
+			errorPrinter.Println(err.Error())
+		}
+	}
+}
+
 func runGet(cmd *cobra.Command, args []string) error {
-	return runTask(cmd, args, func(c client.C1Client, ctx context.Context, appId string, entitlementId string, userId string, justification string, duration string) (*c1api.C1ApiTaskV1Task, error) {
-		accessRequest, err := c.CreateGrantTask(ctx, appId, entitlementId, userId, justification, duration)
+	_, _, v, err := cmdContext(cmd)
+	if err != nil {
+		return err
+	}
+
+	return runTask(cmd, args, func(c client.C1Client, ctx context.Context, appId string, entitlementId string, userId string, justification string) (*c1api.C1ApiTaskV1Task, error) {
+		duration := v.GetString(durationFlag)
+
+		entitlement, err := c.GetEntitlement(ctx, appId, entitlementId)
+		if err != nil {
+			return nil, err
+		}
+
+		// entitlement.DurationGrant is assumed to be nil or a non-zero parsable string
+		durationStr := client.StringFromPtr(entitlement.DurationGrant)
+		var maxProvision *time.Duration
+		maxProvisionTime, err := time.ParseDuration(durationStr)
+		if err == nil {
+			maxProvision = &maxProvisionTime
+		}
+
+		validDuration, err := getValidDuration(ctx, v, maxProvision, duration)
+		if err != nil {
+			return nil, err
+		}
+
+		apiDuration := ""
+		if validDuration != nil {
+			// API expects seconds formated like "1s"
+			seconds := int(validDuration.Seconds())
+			apiDuration = fmt.Sprintf("%ds", seconds)
+		}
+
+		accessRequest, err := c.CreateGrantTask(ctx, appId, entitlementId, userId, justification, apiDuration)
 		if err != nil {
 			openApiError := &c1api.GenericOpenAPIError{}
 			if !errors.As(err, &openApiError) {
 				return nil, err
 			}
 			errorBody := string(openApiError.Body())
 			if strings.Contains(errorBody, grantDurationErrorMessage) {
-				return nil, fmt.Errorf(grantDurationErrorMessage)
+				startIndex := strings.Index(errorBody, grantDurationErrorMessage)
+				endIndex := strings.LastIndex(errorBody, ")") + 1
+				return nil, errors.New(errorBody[startIndex:endIndex])
```

Comment:
What is this removing from the error message?

Reply:
its like all the json stuff so {"message": "..."} gets everything in the ...